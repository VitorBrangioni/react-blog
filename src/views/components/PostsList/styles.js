import styled from "styled-components";

export const Container = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(350px, 1fr));
  width: 100%;
`;
