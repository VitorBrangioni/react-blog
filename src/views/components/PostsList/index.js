import React, { Component } from "react";
import Post from "../Post";
import { Container } from "./styles";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Creators as PostsActions } from "../../../store/ducks/postDuck";
import { PostsListLoader } from "../Loader";

class PostsList extends Component {
  constructor(props) {
    super(props);
    this.props.getPosts(null, -1);
  }

  render() {
    const { data, loading } = this.props.resGetPosts;
    if (loading) {
      return (<PostsListLoader />);
    } 
    return (
      <Container>
        {data.map((post, i) => (
          <Post post={post} key={i} />
        ))}
      </Container>
    );
  }
}

const mapStateToProps = ({ PostDuck }) => {
  return {
    resGetPosts: PostDuck,
  };
};

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(PostsActions, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(PostsList);
