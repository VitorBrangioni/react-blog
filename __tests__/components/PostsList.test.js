import React from "react";
import PostsList from "../../src/views/components/PostsList";
import postsMock from "../../__mocks__/posts.mock";
import { shallow } from "enzyme";
import { render } from "@testing-library/react";

import { createStore } from "redux";

import { Provider } from "react-redux";

const initialState = {
  data: postsMock,
  recentPosts: [],
  authors: [],
  loading: false,
  error: false,
};

const statingState = { data: [], PostDuck: initialState };
function reducer(state = statingState, action) {
  return state;
}

const renderWithRedux = async (
  component,
  { initialState, store = createStore(reducer, initialState) } = {}
) => {
  return {
    ...render(<Provider store={store}>{component}</Provider>),
  };
};

describe("PostsList", () => {
  it("Verify if all posts are rendering", async () => {
    const { queryByTestId } = await renderWithRedux(<PostsList />);

    postsMock.forEach((_, i) => {
      const element = queryByTestId(`post-${i}`);
      expect(element).not.toBeNull();
    });
  });
});
