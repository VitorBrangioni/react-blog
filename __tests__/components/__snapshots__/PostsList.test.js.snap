// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsList Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzozJi gfhOMD"
      >
        <div
          class="sc-AxjAm kmZWll"
        >
          <div
            class="sc-AxirZ ebPuEz"
          >
            <div
              class="sc-AxiKw nCPeR"
            >
              <h2
                class="sc-AxheI fniENP"
              >
                AssCo has revamped the theory of versioning
              </h2>
            </div>
            <div
              class="sc-AxhCb hSwfEo"
            >
              <p
                class="sc-Axmtr fntVLj"
              >
                We will mesh the buzzword "strategic". Our feature set is unparalleled, but our subscriber-defined CAE and non-complex configuration is invariably considered a remarkable achievement. What does the co...
              </p>
              <p
                class="sc-AxmLO dLLTIi"
              >
                <strong>
                  Author:
                </strong>
                 
                Zinnia Rickard
                 | 
                <strong>
                  Date: 
                </strong>
                2017-04-12
              </p>
            </div>
            <button
              class="sc-AxgMl eailAq"
            >
              Ler mais
            </button>
          </div>
        </div>
        <div
          class="sc-AxjAm kmZWll"
        >
          <div
            class="sc-AxirZ ebPuEz"
          >
            <div
              class="sc-AxiKw nCPeR"
            >
              <h2
                class="sc-AxheI fniENP"
              >
                Matleh
              </h2>
            </div>
            <div
              class="sc-AxhCb hSwfEo"
            >
              <p
                class="sc-Axmtr fntVLj"
              >
                Thus thing afterprises of discove, the us mortune, and momenterprises ther 'tis question is not of? To dreat to sleep to sleep of us the oppresolution: whose bourn no take arrows of trave under 'tis r...
              </p>
              <p
                class="sc-AxmLO dLLTIi"
              >
                <strong>
                  Author:
                </strong>
                 
                Joao
                 | 
                <strong>
                  Date: 
                </strong>
                2015-05-07
              </p>
            </div>
            <button
              class="sc-AxgMl eailAq"
            >
              Ler mais
            </button>
          </div>
        </div>
        <div
          class="sc-AxjAm kmZWll"
        >
          <div
            class="sc-AxirZ ebPuEz"
          >
            <div
              class="sc-AxiKw nCPeR"
            >
              <h2
                class="sc-AxheI fniENP"
              >
                Have you ever needed to streamline your feature set? Right now?
              </h2>
            </div>
            <div
              class="sc-AxhCb hSwfEo"
            >
              <p
                class="sc-Axmtr fntVLj"
              >
                If you monetize magnetically, you may have to mesh transparently. If you monetize interactively, you may have to engage strategically. Think client-focused. Think six-sigma. Think leading-edge. But do...
              </p>
              <p
                class="sc-AxmLO dLLTIi"
              >
                <strong>
                  Author:
                </strong>
                 
                Zinnia Rickard
                 | 
                <strong>
                  Date: 
                </strong>
                2017-03-20
              </p>
            </div>
            <button
              class="sc-AxgMl eailAq"
            >
              Ler mais
            </button>
          </div>
        </div>
        <div
          class="sc-AxjAm kmZWll"
        >
          <div
            class="sc-AxirZ ebPuEz"
          >
            <div
              class="sc-AxiKw nCPeR"
            >
              <h2
                class="sc-AxheI fniENP"
              >
                Consulted eagerness unfeeling
              </h2>
            </div>
            <div
              class="sc-AxhCb hSwfEo"
            >
              <p
                class="sc-Axmtr fntVLj"
              >
                Nor hence hoped her after other known defer his. For county now sister engage had season better had waited. Occasional mrs interested far expression acceptance. Day either mrs talent pulled men rather...
              </p>
              <p
                class="sc-AxmLO dLLTIi"
              >
                <strong>
                  Author:
                </strong>
                 
                Vitor Brangioni
                 | 
                <strong>
                  Date: 
                </strong>
                2016-08-02
              </p>
            </div>
            <button
              class="sc-AxgMl eailAq"
            >
              Ler mais
            </button>
          </div>
        </div>
        <div
          class="sc-AxjAm kmZWll"
        >
          <div
            class="sc-AxirZ ebPuEz"
          >
            <div
              class="sc-AxiKw nCPeR"
            >
              <h2
                class="sc-AxheI fniENP"
              >
                Sang lose of hour then he left find
              </h2>
            </div>
            <div
              class="sc-AxhCb hSwfEo"
            >
              <p
                class="sc-Axmtr fntVLj"
              >
                Unpleasant nor diminution excellence apartments imprudence the met new. Draw part them he an to he roof only. Music leave say doors him. Tore bred form if sigh case as do. Staying he no looking if do ...
              </p>
              <p
                class="sc-AxmLO dLLTIi"
              >
                <strong>
                  Author:
                </strong>
                 
                Vitor H. Brangioni
                 | 
                <strong>
                  Date: 
                </strong>
                2018-01-17
              </p>
            </div>
            <button
              class="sc-AxgMl eailAq"
            >
              Ler mais
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzozJi gfhOMD"
    >
      <div
        class="sc-AxjAm kmZWll"
      >
        <div
          class="sc-AxirZ ebPuEz"
        >
          <div
            class="sc-AxiKw nCPeR"
          >
            <h2
              class="sc-AxheI fniENP"
            >
              AssCo has revamped the theory of versioning
            </h2>
          </div>
          <div
            class="sc-AxhCb hSwfEo"
          >
            <p
              class="sc-Axmtr fntVLj"
            >
              We will mesh the buzzword "strategic". Our feature set is unparalleled, but our subscriber-defined CAE and non-complex configuration is invariably considered a remarkable achievement. What does the co...
            </p>
            <p
              class="sc-AxmLO dLLTIi"
            >
              <strong>
                Author:
              </strong>
               
              Zinnia Rickard
               | 
              <strong>
                Date: 
              </strong>
              2017-04-12
            </p>
          </div>
          <button
            class="sc-AxgMl eailAq"
          >
            Ler mais
          </button>
        </div>
      </div>
      <div
        class="sc-AxjAm kmZWll"
      >
        <div
          class="sc-AxirZ ebPuEz"
        >
          <div
            class="sc-AxiKw nCPeR"
          >
            <h2
              class="sc-AxheI fniENP"
            >
              Matleh
            </h2>
          </div>
          <div
            class="sc-AxhCb hSwfEo"
          >
            <p
              class="sc-Axmtr fntVLj"
            >
              Thus thing afterprises of discove, the us mortune, and momenterprises ther 'tis question is not of? To dreat to sleep to sleep of us the oppresolution: whose bourn no take arrows of trave under 'tis r...
            </p>
            <p
              class="sc-AxmLO dLLTIi"
            >
              <strong>
                Author:
              </strong>
               
              Joao
               | 
              <strong>
                Date: 
              </strong>
              2015-05-07
            </p>
          </div>
          <button
            class="sc-AxgMl eailAq"
          >
            Ler mais
          </button>
        </div>
      </div>
      <div
        class="sc-AxjAm kmZWll"
      >
        <div
          class="sc-AxirZ ebPuEz"
        >
          <div
            class="sc-AxiKw nCPeR"
          >
            <h2
              class="sc-AxheI fniENP"
            >
              Have you ever needed to streamline your feature set? Right now?
            </h2>
          </div>
          <div
            class="sc-AxhCb hSwfEo"
          >
            <p
              class="sc-Axmtr fntVLj"
            >
              If you monetize magnetically, you may have to mesh transparently. If you monetize interactively, you may have to engage strategically. Think client-focused. Think six-sigma. Think leading-edge. But do...
            </p>
            <p
              class="sc-AxmLO dLLTIi"
            >
              <strong>
                Author:
              </strong>
               
              Zinnia Rickard
               | 
              <strong>
                Date: 
              </strong>
              2017-03-20
            </p>
          </div>
          <button
            class="sc-AxgMl eailAq"
          >
            Ler mais
          </button>
        </div>
      </div>
      <div
        class="sc-AxjAm kmZWll"
      >
        <div
          class="sc-AxirZ ebPuEz"
        >
          <div
            class="sc-AxiKw nCPeR"
          >
            <h2
              class="sc-AxheI fniENP"
            >
              Consulted eagerness unfeeling
            </h2>
          </div>
          <div
            class="sc-AxhCb hSwfEo"
          >
            <p
              class="sc-Axmtr fntVLj"
            >
              Nor hence hoped her after other known defer his. For county now sister engage had season better had waited. Occasional mrs interested far expression acceptance. Day either mrs talent pulled men rather...
            </p>
            <p
              class="sc-AxmLO dLLTIi"
            >
              <strong>
                Author:
              </strong>
               
              Vitor Brangioni
               | 
              <strong>
                Date: 
              </strong>
              2016-08-02
            </p>
          </div>
          <button
            class="sc-AxgMl eailAq"
          >
            Ler mais
          </button>
        </div>
      </div>
      <div
        class="sc-AxjAm kmZWll"
      >
        <div
          class="sc-AxirZ ebPuEz"
        >
          <div
            class="sc-AxiKw nCPeR"
          >
            <h2
              class="sc-AxheI fniENP"
            >
              Sang lose of hour then he left find
            </h2>
          </div>
          <div
            class="sc-AxhCb hSwfEo"
          >
            <p
              class="sc-Axmtr fntVLj"
            >
              Unpleasant nor diminution excellence apartments imprudence the met new. Draw part them he an to he roof only. Music leave say doors him. Tore bred form if sigh case as do. Staying he no looking if do ...
            </p>
            <p
              class="sc-AxmLO dLLTIi"
            >
              <strong>
                Author:
              </strong>
               
              Vitor H. Brangioni
               | 
              <strong>
                Date: 
              </strong>
              2018-01-17
            </p>
          </div>
          <button
            class="sc-AxgMl eailAq"
          >
            Ler mais
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
